packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -56,10 +54,12 @@ class SizeAnalyzer {
     );
 
     // Convert an AOT snapshot file into a map.
-    final Map<String, dynamic> processedAotSnapshotJson = treemapFromJson(
-      json.decode(aotSnapshot.readAsStringSync()),
-    );
-    final _SymbolNode aotSnapshotJsonRoot = _parseAotSnapshot(processedAotSnapshotJson);
+    final Object? decodedAotSnapshot = json.decode(aotSnapshot.readAsStringSync());
+    if (decodedAotSnapshot == null) {
+      throwToolExit('AOT snapshot is invalid for analysis');
+    }
```

Comment:
I can't imagine this ever being null (more likely a `FormatException`) but just in case.